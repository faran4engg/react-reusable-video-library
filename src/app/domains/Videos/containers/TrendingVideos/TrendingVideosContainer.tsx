import { FC, memo } from 'react';
import { OwnProps, RenderProps } from './types';
import {
  useApprovedVideos,
  useTrendingVideoIds,
} from 'app/hooks/queries-hooks';
import { getTrendingVideos } from 'utils/getTrendingVideos';

const TrendingVideosContainer: FC<OwnProps & RenderProps> = ({ children }) => {
  const { isLoading, data: approvedVideos, isSuccess } = useApprovedVideos();
  const { data: videoIdArr } = useTrendingVideoIds({ shouldFetch: isSuccess });

  return (
    <div className="text-gray-600 dark:text-gray-400">
      {children({
        isLoading,
        // @ts-ignore
        videos: getTrendingVideos(videoIdArr?.data, approvedVideos?.data),
      })}
    </div>
  );
};

export default memo(TrendingVideosContainer);
