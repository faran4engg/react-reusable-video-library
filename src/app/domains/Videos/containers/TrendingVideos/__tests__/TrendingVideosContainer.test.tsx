import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import TrendingVideosContainer from '..';

jest.mock('react-query', () => ({
  useQuery: () => ({}),
}));

describe('<TrendingVideosContainer />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render and match the snapshot when loading', () => {
    const renderer = createRenderer();

    renderer.render(
      <TrendingVideosContainer>{() => <div />}</TrendingVideosContainer>,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render and match the snapshot when error', () => {
    const renderer = createRenderer();

    renderer.render(
      <TrendingVideosContainer>{() => <div />}</TrendingVideosContainer>,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
  it('should render and match the snapshot successfully', () => {
    const renderer = createRenderer();
    renderer.render(
      <TrendingVideosContainer>{() => <div />}</TrendingVideosContainer>,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
