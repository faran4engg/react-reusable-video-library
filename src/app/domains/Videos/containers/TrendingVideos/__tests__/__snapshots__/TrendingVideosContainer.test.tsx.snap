// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrendingVideosContainer /> should render and match the snapshot successfully 1`] = `
<div
  className="text-gray-600 dark:text-gray-400"
>
  <div />
</div>
`;

exports[`<TrendingVideosContainer /> should render and match the snapshot when error 1`] = `
<div
  className="text-gray-600 dark:text-gray-400"
>
  <div />
</div>
`;

exports[`<TrendingVideosContainer /> should render and match the snapshot when loading 1`] = `
<div
  className="text-gray-600 dark:text-gray-400"
>
  <div />
</div>
`;
